docs/scraping-receivers.md

```diff
@@ -0,0 +1,143 @@
+the metric and resource attribute names MUST be released in an [OpenTelemetry 
+Specification](https://github.com/open-telemetry/opentelemetry-specification)
+
+## Changing the emitted metrics
+
+Some changes are not considered breaking and can be applied to metrics emitted by scraping receivers of any 
+stability level:
+
+- Adding a new optional metric.
+- Adding a new resource attribute to existing metrics.
+
+Most of other changes to the metrics emitted by scraping receivers are considered breaking and MUST be handled
+according to the stability level of the receiver. Each type of breaking change defines a set of steps that MUST (or 
+SHOULD) be applied across several releases for a Stable (or Beta) components. At least 2 versions SHOULD be kept 
+between the steps to give users time to prepare, e.g. if the step one is released in v0.62.0, the step two should be 
+released in 0.65.0. Any warnings SHOULD include the version starting from which the next step will take action. If 
+breaking change is more complicated and many metrics are involved in the change, feature gates SHOULD be used instead.
+
+### Removing an optional metric
+
+Steps to remove an optional metric:
+
+1. Mark the metric as deprecated in `metadata.yaml` by adding "[DEPRECATED]" in its description. Show a warning that 
+   the metric will be removed if the `enabled` option is set explicitly to `true` in user settings. 
+2. Remove the metric.
+
+### Removing a default metric
+
+Steps to remove a default metric:
+
+1. Mark the metric as deprecated in `metadata.yaml` by adding "[DEPRECATED]" in its description. Show a warning that
+   the metric will be removed if the `enabled` option is not set explicitly to `false` in user settings.
+2. Turn the metric into optional. Show a warning that the metric will be removed if the `enabled` option is set 
+   explicitly to `true` in user settings.
+3. Remove the metric.
+
+### Replacing an optional metric
+
+Steps to replace an optional metric with another optional metric:
+
+1. Mark the metric as deprecated in `metadata.yaml` by adding "[DEPRECATED]" in its description. Add a warning that 
+   the metric will be replaced if the `enabled` option is set explicitly to `true` in user settings. Add the new
+   optional metric in `metadata.yaml`. 
+2. Remove the old metric.
+
+### Replacing a default metric
+
+Steps to replace a default metric with another default metric:
+
+1. Mark the metric as deprecated in `metadata.yaml` by adding "[DEPRECATED]" in its description. Show a warning 
+   that the metric will be replaced if the `enabled` option is not set explicitly to `false` in user settings. Add the
+   new metric as optional for now. Show another warning for the new metric if the `enabled` option is not set 
+   explicitly to any value in user settings.
+2. Turn the old metric into optional. Turn the new metric into default. Show the warning in case if user keeps the 
+   old metric enabled explicitly.
+3. Remove the old metric.
```

Comment:
If a metric is being replaced with another, I think we should just refer to "Removing a default metric" and "Adding a default metric"?

That said, I think we should consider whether there is a reasonable process for making breaking changes to a metric (without changing its name). Often, a metric name is very intuitive and really seems like the "right" name for that metric. I think we're creating an undesirable situation if metric names are burned for good due to bugs in the data model that are resolved during beta.

Reply:
Makes sense. Thanks for the feedback. I removed the "replacement" sections to not encourage unnecessary renames. Also added a section for other in-place changes using feature gates.